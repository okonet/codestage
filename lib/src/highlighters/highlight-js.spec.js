/* eslint no-unused-expressions: 0 */
const path = require('path')
const {
  highlight,
  getThemes,
  getLanguages,
  resolvePackageDir,
  loadTheme
} = require('./highlight-js')

describe('highlight', () => {
  it('should return HTML with proper class names', async () => {
    const html = await highlight('async function html2rtf() {}', 'javascript')
    expect(html).toMatchSnapshot()
  })
})

describe('resolvePackageDir', () => {
  it('should resolve path to highlight.js/styles dir', () => {
    expect(resolvePackageDir('highlight.js')).toEqual(
      path.join(__dirname, '..', '..', 'node_modules', 'highlight.js')
    )
  })
})

describe('loadTheme', () => {
  it('should load highlight CSS and return a string', async () => {
    const { cssText } = await loadTheme('agate')
    expect(cssText).toMatchSnapshot()
  })

  it('should match the shape', async () => {
    const theme = await loadTheme('github')
    expect(theme).toEqual(
      expect.objectContaining({
        cssText: expect.any(String),
        name: expect.any(String)
      })
    )
  })
})

describe('getThemes', () => {
  it('should return themes as a key values object', async () => {
    const themes = await getThemes()
    Object.keys(themes).forEach(key => expect(key).toEqual(themes[key].name))
  })

  it('should match the shape', async () => {
    const themes = await getThemes()
    Object.values(themes).forEach(theme => {
      expect(theme).toEqual(
        expect.objectContaining({
          cssText: expect.any(String),
          name: expect.any(String),
          path: expect.any(String)
        })
      )
    })
  })
})

describe('getLanguages', () => {
  it('should return array of objects with names and paths to languages', async () => {
    expect(await getLanguages()).toMatchSnapshot()
  })
})
