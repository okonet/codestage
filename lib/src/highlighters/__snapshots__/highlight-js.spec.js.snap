// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLanguages should return array of objects with names and paths to languages 1`] = `
Array [
  "1c",
  "abnf",
  "accesslog",
  "actionscript",
  "ada",
  "apache",
  "applescript",
  "arduino",
  "armasm",
  "asciidoc",
  "aspectj",
  "autohotkey",
  "autoit",
  "avrasm",
  "awk",
  "axapta",
  "bash",
  "basic",
  "bnf",
  "brainfuck",
  "cal",
  "capnproto",
  "ceylon",
  "clean",
  "clojure-repl",
  "clojure",
  "cmake",
  "coffeescript",
  "coq",
  "cos",
  "cpp",
  "crmsh",
  "crystal",
  "cs",
  "csp",
  "css",
  "d",
  "dart",
  "delphi",
  "diff",
  "django",
  "dns",
  "dockerfile",
  "dos",
  "dsconfig",
  "dts",
  "dust",
  "ebnf",
  "elixir",
  "elm",
  "erb",
  "erlang-repl",
  "erlang",
  "excel",
  "fix",
  "flix",
  "fortran",
  "fsharp",
  "gams",
  "gauss",
  "gcode",
  "gherkin",
  "glsl",
  "go",
  "golo",
  "gradle",
  "groovy",
  "haml",
  "handlebars",
  "haskell",
  "haxe",
  "hsp",
  "htmlbars",
  "http",
  "hy",
  "inform7",
  "ini",
  "irpf90",
  "java",
  "javascript",
  "jboss-cli",
  "json",
  "julia-repl",
  "julia",
  "kotlin",
  "lasso",
  "ldif",
  "leaf",
  "less",
  "lisp",
  "livecodeserver",
  "livescript",
  "llvm",
  "lsl",
  "lua",
  "makefile",
  "markdown",
  "mathematica",
  "matlab",
  "maxima",
  "mel",
  "mercury",
  "mipsasm",
  "mizar",
  "mojolicious",
  "monkey",
  "moonscript",
  "n1ql",
  "nginx",
  "nimrod",
  "nix",
  "nsis",
  "objectivec",
  "ocaml",
  "openscad",
  "oxygene",
  "parser3",
  "perl",
  "pf",
  "php",
  "pony",
  "powershell",
  "processing",
  "profile",
  "prolog",
  "protobuf",
  "puppet",
  "purebasic",
  "python",
  "q",
  "qml",
  "r",
  "rib",
  "roboconf",
  "routeros",
  "rsl",
  "ruby",
  "ruleslanguage",
  "rust",
  "scala",
  "scheme",
  "scilab",
  "scss",
  "shell",
  "smali",
  "smalltalk",
  "sml",
  "sqf",
  "sql",
  "stan",
  "stata",
  "step21",
  "stylus",
  "subunit",
  "swift",
  "taggerscript",
  "tap",
  "tcl",
  "tex",
  "thrift",
  "tp",
  "twig",
  "typescript",
  "vala",
  "vbnet",
  "vbscript-html",
  "vbscript",
  "verilog",
  "vhdl",
  "vim",
  "x86asm",
  "xl",
  "xml",
  "xquery",
  "yaml",
  "zephir",
]
`;

exports[`highlight should return HTML with proper class names 1`] = `"<div class=\\"hljs\\"><span class=\\"hljs-keyword\\">async</span> <span class=\\"hljs-function\\"><span class=\\"hljs-keyword\\">function</span> <span class=\\"hljs-title\\">html2rtf</span>(<span class=\\"hljs-params\\"></span>) </span>{}</div>"`;

exports[`loadTheme should load highlight CSS and return a string 1`] = `
"/*!
 * Agate by Taufik Nurrohman <https://github.com/tovic>
 * ----------------------------------------------------
 *
 * #ade5fc
 * #a2fca2
 * #c6b4f0
 * #d36363
 * #fcc28c
 * #fc9b9b
 * #ffa
 * #fff
 * #333
 * #62c8f3
 * #888
 *
 */.hljs{display:block;overflow-x:auto;padding:0.5em;color:white}.hljs-name,.hljs-strong{font-weight:bold}.hljs-code,.hljs-emphasis{font-style:italic}.hljs-tag{color:#62c8f3}.hljs-variable,.hljs-template-variable,.hljs-selector-id,.hljs-selector-class{color:#ade5fc}.hljs-string,.hljs-bullet{color:#a2fca2}.hljs-type,.hljs-title,.hljs-section,.hljs-attribute,.hljs-quote,.hljs-built_in,.hljs-builtin-name{color:#ffa}.hljs-number,.hljs-symbol,.hljs-bullet{color:#d36363}.hljs-keyword,.hljs-selector-tag,.hljs-literal{color:#fcc28c}.hljs-comment,.hljs-deletion,.hljs-code{color:#888}.hljs-regexp,.hljs-link{color:#c6b4f0}.hljs-meta{color:#fc9b9b}.hljs-deletion{background-color:#fc9b9b;color:#333}.hljs-addition{background-color:#a2fca2;color:#333}.hljs a{color:inherit}.hljs a:focus,.hljs a:hover{color:inherit;text-decoration:underline}"
`;
